File: java_pi_dir/pi.java
public class pi {
    public static void main(String[] argc) {
        long i, numsteps, start, stop, difference;
        double step, x, psum, mypi, time;

        numsteps = 500000000;

        if (argc.length > 0) {            
            numsteps = Integer.parseInt(argc[0]);
        }
        
        System.out.println("Calculating PI using:");
        System.out.println("  " + numsteps + " slices");
        System.out.println("  1 processes");

        start = System.currentTimeMillis();

        psum = 0.0d;
        step = 1d/(double)numsteps;
        
        for (i = 0; i < numsteps; i++) {
            x = ((double)i + 0.5d) * step;
            psum = psum + 4d/(1d + (x * x));
        } 
        
        mypi = psum * step;
        
        stop = System.currentTimeMillis();
        difference = stop - start;
        time = (double)(difference)/1000d;
        
        System.out.println("Obtained value of PI: " + mypi);
        System.out.println("Time taken: " + time + " seconds");

    }
}

File: clojure_pi_dir/pi.clj
; We have to define name-space etc. because we want to compile to to Java byte-code.
(ns pi
  (:gen-class
   :methods [[calcpi [] Integer]]))

; The method that does the bulk of our work as usual.
(defn calcpi [n]
  (println "Calculating PI using:")
  (println " " n "slices")
  (println "  1 processes")

; When referring to Java stuff the syntax is class/method.
  (def start (System/currentTimeMillis))
  (def psum 0.0) 
  (def step (/ 1 n))
  
  (loop [i 0] (when (< i n)
    (def x (* (+ i 0.5) step))
    (def psum (+ psum (/ 4 (+ 1 (* x x)))))
    (recur (inc i))
  ))

  (def mypi (* psum step))
  (def stop (System/currentTimeMillis))
  (def difference (/ (- stop start) 1000.0))

  (println "Obtained value of PI:" mypi)
  (println "Time taken:" difference "seconds")
)
